mutation/mutation_compactor.hh

```diff
@@ -468,8 +468,16 @@ public:
             _partition_limit -= _rows_in_current_partition > 0;
             auto stop = consumer.consume_end_of_partition();
             if (!sstable_compaction()) {
-                return _row_limit && _partition_limit && stop != stop_iteration::yes
+                stop = _row_limit && _partition_limit && stop != stop_iteration::yes
```

Comment:
This code is confusing. bool(_row_limit) == true doesn't mean we reached the row limit, it means we haven't reached it. Perhaps _rows_remaining is a better name. There are two variables named "stop", they should have different names.

Reply:
Yes, I always have to remind myself how these limits work.